lighthouse-core/lib/dependency-graph/simulator/simulator.js

```diff
@@ -132,16 +139,17 @@ class Estimator {
   }
 
   /**
-   * Initializes the various state data structures such as _nodesInQueue and _nodesCompleted.
+   * Initializes the various state data structures such as _nodesReadyToStart and _nodesCompleted.
    */
   _initializeAuxiliaryData() {
     this._nodeTiming = new Map();
-    this._nodesUnprocessed = new Set();
-    this._nodesCompleted = new Set();
-    this._nodesInProgress = new Set();
-    this._nodesInQueue = new Set(); // TODO: replace this with priority queue
     this._connectionsInUse = new Set();
     this._numberInProgressByType = new Map();
+
+    this._nodes = {};
```

Comment:
should be an array now?

Reply:
wait why? I thought the whole point was to key these by their state 😆

Reply:
> EDIT: oh just that the keys are numbers now so you'd prefer to have array?

yah

Reply:
done